@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -939,20 +949,37 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                             );
                         }
 
+                        if (isSpherical) {
+                            normal
+                                .set(vertices[i], vertices[i + 1], vertices[i + 2])
+                                .add(this.center)
+                                .normalize();
+                        }
+
+                        const minAxis = normal.clone().multiplyScalar(minHeight * scaleFactor);
+                        const maxAxis = normal.clone().multiplyScalar(height * scaleFactor);
                         positions.push(
-                            vertices[i],
-                            vertices[i + 1],
-                            vertices[i + 2] + minHeight * scaleFactor
+                            vertices[i] + minAxis.x,
+                            vertices[i + 1] + minAxis.y,
+                            vertices[i + 2] + minAxis.z
                         );
                         if (texCoordType !== undefined) {
                             textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
                         }
 
                         if (isExtruded) {
                             positions.push(
-                                vertices[i],
-                                vertices[i + 1],
-                                vertices[i + 2] + height * scaleFactor
+                                vertices[i] + minAxis.x,
```

Comment:
As discussed: Please create the fully extruded mesh here and reverse the extrusion in the shader.